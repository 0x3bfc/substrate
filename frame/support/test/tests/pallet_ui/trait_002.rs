#[frame_support::pallet]
mod pallet {
	use frame_support::pallet_prelude::ModuleInterface;
	use frame_system::pallet_prelude::BlockNumberFor;

	#[pallet::trait_]
	pub trait Trait: frame_system::Trait {
		#[pallet::constant]
		type U;
	}

	#[pallet::module]
	pub struct Module<T>(core::marker::PhantomData<T>);

	#[pallet::module_interface]
	impl<T: Trait> ModuleInterface<BlockNumberFor<T>> for Module<T> {}

	#[pallet::call]
	impl<T: Trait> Module<T> {}
}

fn main() {
}
