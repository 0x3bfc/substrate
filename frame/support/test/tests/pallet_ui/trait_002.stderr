error: Invalid usage of `#[pallet::constant]`, syntax must be `type $SomeIdent: Get<$SomeType>;`
 --> $DIR/trait_002.rs:9:3
  |
9 |         type U;
  |         ^^^^

error: expected `:`
 --> $DIR/trait_002.rs:9:9
  |
9 |         type U;
  |               ^
